### ISSUE TITLE
Status of the project

### ISSUE THREAD

**Issue by Svenskunganka** [stylus/stylus#2282]
I'm opening this issue to serve as a place for discussion about the project because I feel like development of the project and its ecosystem has stagnated. There hasn't been a new release in almost a year, there are many confirmed bugs in the issue tracker and most of the PRs are either waiting to be review or are ready to be merged.  
Development of parts of the Stylus ecosystem has more or less stopped too. Many useful plugins and editor integrations has either been put on hold or completely abandoned.

How does it look like for you @Panya, @kizu & @corysimmons? I'm sure you all are busy, but it would be great if we could work out a plan for continued development of Stylus.  
I'm sure there are some community members that would be willing to help out reviewing PRs, sorting out issues and fixing bugs, me included. How does that sound?

Thanks for taking time reading!

**Comment by corysimmons** [comment]
Most of the commits on the core project are from @Panya and @kizu -- I just did the website. I'm happy to update the website when I get some time.

**Comment by kaisermann** [comment]
Willing to help if needed :)

**Comment by corysimmons** [comment]
@kaisermann You prefer [JS](https://github.com/stylus/stylus/issues) or [web-design](https://github.com/stylus/stylus-lang.com/issues) tasks?

The docs should also adhere to a specific convention rather than a confusing grab-bag of possible syntaxes, if you're just up for doing grunt work to make the project stronger.

**Comment by kaisermann** [comment]
I think I prefer the Js tasks :)

**Comment by Svenskunganka** [comment]
I'm down to do either, but I think the best course of action for the time being is adding more members to the org with commit access to the repositories if the current core members are having trouble finding time to set away so that we can start reviewing & merging PRs and grunt out some of the long-standing bugs. However I'm waiting for a comment from @Panya and @kizu and hear out what their opinions and current status looks like.

**Comment by imiuka** [comment]
I'd also love to contribute and see all our big pain points and issues resolved.

I use Stylus extensively, but I personally know of several situations where teams chose Sass over Stylus because of long standing issues which haven't been fixed (even when PRs exist).

**Comment by corysimmons** [comment]
@Svenskunganka I think you, or anyone, would need to contribute some to the project to get in the org.

Just start picking off issues then ask.

This is the way my fathers and their fathers before them have always done it.

**Comment by Svenskunganka** [comment]
@corysimmons Yes of course, that was my plan all along until I noticed the current list of PRs and identified bugs were at a standstill so it seemed pointless to just add to the pile, so to speak.

The reason I opened this issue was to serve as a place to discuss the project and what can be done to kickstart development. I apologize if my previous comment sounded like a takeover, I realize now my bad choice of wording with _"That way, **we** can start [...]"_ might've suggested that, but I personally am not interested in being a member of the org unless absolutely necessary (i.e nobody else can/wants and the current members don't see any change in the foreseeable future), as I cannot myself guarantee I will have the time to set away to manage this project.  
However, I definitely want to and intend to contribute to the project, but the reason I haven't done so thus far is because I'm worried it'll just pile up and go to waste (since there's no one reviewing or merging).

But I agree with you, and will start familliarize myself with the codebase 😄

**Comment by corysimmons** [comment]
@Svenskunganka I feel ya. I hate it when PRs sit dormant.

I don't work on the main project (just the website), but it looks like there are only 16 outstanding PRs with half of them only being a month or 2 old.

Compared to other projects of this size, that ain't too bad.

Your concerns are valid though. I'd love to see this project revitalized. @Panya has been single-handedly working on it for a long time.

That said, I still think the only way Panya (or Kizu, or anyone) is going to trust anyone to merge code is by demonstrating you have the ability to write good code.

0. Write good, non-breaking, correctly versioned, PRs.
0. Work with @Panya and @kizu to make sure they're concerns are addressed about specific PRs.
0. Get a couple merged.
0. Ask to be a contributor.

I know you're on the fence about becoming a contrib, but there were some other people in this thread who mentioned they would be interested. That's the path forward for them... in any project.

**Comment by imiuka** [comment]
i don't think anyone here expects to be randomly given contributor status, especially so without having first made significant contributions to stylus.

at least for me, i was hoping that we would see commitment to people (whether it be old core members, or new, trusted people appointed) to review new issues and pull requests that are filed against stylus.

-----

@Svenskunganka expressed my concerns very well: as an outsider, it looked like stylus was bordering abandonware for a long time. there appeared to be very little activity from core members. and without that, stylus could not move forward. let me explain why:

when i was about to submit my pull request last december, i'd wager that at least half of all issues and pull requests had no comment from any team member at all, even ones that had sat for 6 months. some of my biggest issues and pain points with stylus had been sitting there, seemingly dead, without comment, merge, or rejection, for a good half year or year.

there'd been no action to acknowledge that yes, some issue was a bug, or to merge (or reject) a given pull request, or anything of that sort. the commit log did not give any more inspiration, either. the project looked like it'd sat dead for a good year, and any changes one might submit would go unheard.

i think that's been the factor driving people towards alternatives, and from trying to fix issues - your efforts seemed to go down an empty vortex. i see now that the `pr: needs review` tag has been added to some of them, though no review seems to be taking place.

so in summary, at least several of us are just hoping that if we spend time writing an improvement or bug fix, someone will say "this is bad, i won't merge it", or "thank you" and then actually hit the merge button. similarly for issues.

i love stylus, i'd love to see a future where stylus is still commonplace ♡

**Comment by corysimmons** [comment]
I didn't look at the PRs. I didn't realize things had sat dormant for so long. Panya always seemed fast to me.

I think @Panya @kizu are the only ones that can add someone else. It's entirely up to them.

>i love stylus, and i desperately hope to see a future where stylus is still commonplace ♡

Me too.

**Comment by Svenskunganka** [comment]
@corysimmons You are right when you say that the issue tracker & the PRs are still well organized, but I still think it is a good idea to take care of the problem while the community around Stylus is still healthy. I think most people react negatively to a project not having a commit or release in a year, let alone two or three if we were to wait that long to consider the project "abandoned".

Ideally I wanted to hear from @Panya or @kizu about how their status looks like and what they're thinking before we got into serious discussion about what to do, and I wanted to wait a month or two for a response instead of jumping straight into it so we don't stress them in their most likely already stressed situation.  
Perhaps they're just temporarily on a tight schedule and are planning resume as usual when it clears up, who knows?

Anyway, I'm going to pick off some of the easier bugs, starting with #2284. The codebase is as you say huge, so it is going to take a while to fully familiarize myself.

**Comment by corysimmons** [comment]
Could also do something like io.js did when the Node team was dragging. Fork and merge anything you want. If the fork is passing tests, looks good, and is picking up steam, then the 2 projects could be merged in the future.

**Comment by chrisdothtml** [comment]
Just noticed this issue and love Stylus. Would be happy to contribute where needed

**Comment by maranomynet** [comment]
Stylus has gone through two spurts of activity. https://github.com/stylus/stylus/graphs/contributors?from=2011-05-16&to=2017-04-19&type=c

The first one during the first 18 months or so when @tj was pretty much doing everything. Then when he abandoned the project mid-year 2012 there was about a year of nothing happening, until @Panya and @kizu took over early summer 2013 and started the second spurt which lasted until early 2016.

Question is, are there people willing to step up and assume the role of lead developers, and will the current maintainers let them?

**Comment by maranomynet** [comment]
I feel like leaving a I❤️Stylus appreciation comment to cheer this thread up a bit...

**Things I love about Stylus:**

* Strong focus on the usability for the CSS developers.
* Friendly developers, willing to discuss and embrace new ideas.
* Ability to read data from JSON files `json()`
* Uncluttered, readable syntax with minimal verbosity
* Dead easy extensibility via `use()` and JavaScript modules.
* Transparent mixins

**Minor peeves:**

* `-` is allowed as first character of variable/mixin/function names
* Using `calc()` is cumbersome

**Comment by chrisdothtml** [comment]
@maranomynet My favorite thing about it has always been how close the syntax is to Javascript. LESS feels like CSS with a few tools draped over it with less-than-desirable syntax. Stylus has always felt like an actual full CSS programming language. It's why I still prefer it over things like PostCSS

**Comment by corysimmons** [comment]
>It's why I still prefer it over things like PostCSS
PostCSS is dank, but SugarSS isn't close to Stylus/Sass/LESS. I'd love to just see tighter integration with Stylus (day-to-day writing) + PostCSS (cross-processor/superpowerful libs).

**Comment by kizu** [comment]
Hi, everyone.

First of all, ❤️  for all of you who care about the project!

As Stylus' maintainer I'm the one responsible for the project's current stage and stagnation. Due to _reasons_, I wasn't able to contribute much to Stylus (and almost anything else in open source community), and couldn't keep on maintaining Stylus as I was before. For a long time @Panya had taken most of my maintainer duties and done what he could, and that was very generous of him, but not everyone have the energy, will, time and motivation to work in their free time on such a project. If he'd want, he'll comment there too, I don't know if he have plans to keep contributing or not, but Stylus clearly needs not just one developer and maintainer for it to live happily without stagnation.

We had some kind of roadmap to 1.0, but I'm still not sure we went in a right direction. At current state I don't know if we could do what we wanted to do for it, and maybe it should be reevaluated and 1.0 should be just the current version with much less new features and syntax changes that we had planned. But, again, even for that we'd need people who could take their time to understand the old rusty codebase, to write new code and to maintain things.

Personally, I don't know if I can be such person. I'd want to, but I don't the have any energy left. Maybe things would change for me in the few coming months (hopefully!), maybe not.

I'll try to find time to look at current PRs and latest issues and see if it is possible to at least come up with a fresh patch with the latest stuff, but I can't promise I'll keep with it.

So, if someone have time and all what it takes, they can join in reviewing the PRs, issues, answer questions and writing new code. It is not necessary should be just one person: everyone could do what they can, and maybe together you could create a team that could pick up Stylus and freshen it a bit? That would be nice.

Again, ❤️, and you can write me here or at kizmarh@ya.ru if you'd see yourself as someone who'd like to be a maintainer and/or developer for Stylus. I'll try to respond when I can.

**Comment by corysimmons** [comment]
Thanks for all you and Panya have done. Stylus is a lot of code for 1 or 2 people to maintain on their own for so long...

**Comment by tomhodgins** [comment]
Massive thanks to all developers, contributors, and maintainers of Stylus. This project will continue be a useful tool, and serve as an inspiration for writing CSS for a _long time_ to come, even if little or not development happens after today.

I think they say “Once you hit max level… you stop levelling”, it looks like Stylus has reached “max level” for now :D

![](http://i.imgur.com/EWqtL.jpg)
^ Stylus now, and Stylus in the future

**Comment by ruslanchek** [comment]
> it looks like Stylus has reached “max level” for now :D

If so, why it's level stuck on 0.54.5?

**Comment by atilkan** [comment]
A beautiful language left for dead. Shame.

**Comment by corysimmons** [comment]
Open source sucks dude. It doesn't pay anything and takes time people could devote to hobbies and their health.

People like TJ have donated more than enough of their free time to making others' lives a bit easier.

Projects like Gratipay (now defunct), Bountysource, and OpenCollective all tried to sprout up as a way to fund these angels who make our lives as developers less frustrating, but the generosity of other developers and million-dollar companies (who are built on these free technologies) is virtually non-existent.

Relevant example: TJ tried to get donation support for [Apex](https://opencollective.com/apex) (making AWS Lambda accessible to everyone) and ended up with $1k/year so he turned it into a legit business with pricing tiers and all that. Now he can afford pet food.

Open source isn't sustainable. It's a meme big corporations are cramming down peoples throats because it generates slave labor for them.

The only relevant open source anymore are these overengineered frameworks Google and Facebook release every now and then.

Stop _not-so-subtly_ pinging people to make your life easier. Instead, dump Stylus/LESS/Sass/PostCSS. Learn React + styled-components. Get money. Live frugal. Save. Retire young. Enjoy life.

The artsy/beautiful internet died with Geocities! Long live this neverending phantasmagoric McCapitalist nightmare! >:p

![image](https://user-images.githubusercontent.com/224636/44445661-d703a400-a5af-11e8-994b-99d115329149.png)

**tl;dr** SugarSS is essentially a modern and maintained version of Stylus. :)

**Comment by mrG0bliN** [comment]
Open source

**Comment by arty-name** [comment]
Would it make sense to link to this issue from the readme? Around the top maybe. This might inspire more people to contribute.

**Comment by mihaon** [comment]
How can I donate to the project?

**Comment by iChenLei** [comment]
Hi , everyone. I am new maintainer of stylus. Stylus was born 11 years ago, w3c and csswg create and deprecate many new css specs with the passage of time. Stylus didn't follow css standard as soon as possible, so lack modern css features support(e.g. [Media Queries Level 4](https://www.w3.org/TR/mediaqueries-4), [css-nesting](https://drafts.csswg.org/css-nesting/), [Selectors Level 4](https://www.w3.org/TR/selectors-4) and more ). So I try do some tiny works for stylus iteration.

1. Update test CI infra (This is the basis for development) via https://github.com/stylus/stylus/pull/2601
2. Open [Stylus Github Discussions](https://github.com/stylus/stylus/discussions)
3. Add issue and pull request template https://github.com/stylus/stylus/pull/2606
2. Release new version of vscode-stylus (for IDE/Editor support, this is important for DX) via https://github.com/d4rkr00t/language-stylus/pull/84 [![VSCode Marketplace Installs](https://img.shields.io/visual-studio-marketplace/i/sysoev.language-stylus)](https://marketplace.visualstudio.com/items?itemName=sysoev.language-stylus)

You can't imagine that I do many work for superstar opensource project ([less](https://github.com/less/less.js)/[reactjs](https://github.com/facebook/react)/[mobxjs](https://github.com/mobxjs/mobx)/[vetur](https://github.com/vuejs/vetur)/[emotionjs](https://github.com/emotion-js/emotion) and more) and even though I work at a e-commerce company which must work six days per week. What a crazy world. I don't need any donatation currently, work only for love ❤️ , the love for open source world.

-----------------

Q: Did you mean this project restart development ?
A: Yes, but no promise.
